pkg/build/api/validation/validation.go

```diff
@@ -239,7 +240,8 @@ func validateImageChange(imageChange *buildapi.ImageChangeTrigger) errs.Validati
 	if len(imageChange.Image) == 0 {
 		allErrs = append(allErrs, errs.NewFieldRequired("image"))
 	}
-	if len(imageChange.From.Name) == 0 {
+	var emptyObject kapi.ObjectReference
+	if imageChange.From == emptyObject {
```

Comment:
@mfojtik I have changed it back to the original fix which I find more sane until someone decides that we have to change object references to pointers.